modules/ROOT/pages/dataweave-maven-plugin.adoc

```diff
@@ -0,0 +1,318 @@
+|`boolean`
+|`false`
+|When set to `true`, the test suite generates a coverage report about the execution.
+
+|`coverageFormat`
+|`String`
+|`sonar`
+|Specifies the format for the coverage report, one of:
+
+* `sonar` +
+Generates a valid https://www.sonarqube.org/[Sonar^] coverage report for your test suite.
+* `html` +
+Generates an HTML coverage report for your test suite.
+
+|`runnerLogForkedProcessCommand`
+|`boolean`
+|`false`
+|When set to `true`, logs the command used to launch the test runner process.
+
+|`runnerEnvironmentVariables`
+|`Map`
+|`null`
+|A set of additional environment variables to pass to the test runner process.
+
+|`runnerSystemProperties`
+|`Map`
+|`null`
+|A set of additional system properties to pass to the test runner process.
+
+|`runnerArgLine`
+|`boolean`
+|`null`
+|A set of additional JVM options to pass to the test runner process.
+
+|`runnerJvmDebug`
+|`boolean`
+|`false`
+|When set to `true`, enables remote debugging for the test runner JVM.
+|===
+
+=== Test Configuration Example
+
+The following example enables the `sonar` coverage report and disables `htmlReport` in a project.
+
+.Example `pom.xml` file
+[source,xml,linenums]
+----
+<plugin>
+include::dataweave::partial$dataweave-maven-plugin-config.adoc[tag=config]
+  <configuration>
+    ...
+    <tests>
+      <htmlReport>false</htmlReport>
+      <coverageEnabled>true</coverageEnabled>
+      <coverageFormat>sonar</coverageFormat>
+    </tests>
+    ...
+  </configuration>
+</plugin>
+----
+
+== Generate Documentation for a DataWeave Library
+
+Run the `data-weave:generate-docs` goal to auto-generate the documentation for your DataWeave library, for example:
+
+[source,console,linenums]
+----
+mvn prepare-package
+----
+
+Alternatively, you can also run:
+
+[source,console,linenums]
+----
+mvn data-weave:generate-docs
+----
+
+=== Configure the Automated Docs Generation
+
+Inside the `plugin` configuration, add a `docs` element configured with parameters that meet your desired documentation needs:
+
+[%header%autowidth.spread,cols=".^a,.^a,.^a,.^a"]
+|===
+|Name |Type |Default | Description
+
+|`output`
+|`String`
+|`${project.build.directory}/data-weave-docs`
+|The output directory in which the documentation files are created.
+
+|`template`
+|`String`
+|`exchange_markdown`
+|The template to use for the auto-generated documentation, one of:
+
+* `exchange_markdown` +
+Generates the documentation in a Markdown format that is compliant with https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/exchange-experience-api/[Exchange^].
+* `markdown` +
+Generates the documentation in Markdown format.
+* `asciidoc` +
+Generates the documenation in AsciiDoc format.
+
+Valid values are `exchange_markdown`, `markdown`, or `asciidoc`.
+
+|`homePage`
+|`String`
+|`null`
+|Specifies the path to a custom Markdown file to add in the home page generated by the `markdown` or `exchange_markdown` template.
```

Comment:
Could be read to say that the home page is generated by the template when it is the file that's generated.